tests/foreman/cli/test_usergroup.py

```diff
@@ -313,78 +305,66 @@ def test_negative_automate_bz1437578(self):
+class TestFreeIPAUserGroup:
     """Implements FreeIPA LDAP feature tests for user groups in CLI."""
 
-    @classmethod
     @skip_if_not_set('ipa')
-    def setUpClass(cls):
+    @pytest.fixture(scope='module')
+    def module_creds(self):
+        ldap_user_name = settings.ipa.username_ipa
+        ldap_user_passwd = settings.ipa.password_ipa
+        return {'username': ldap_user_name, 'password': ldap_user_passwd}
+
+    @skip_if_not_set('ipa')
+    @pytest.fixture(scope='module')
+    def module_auth(self, module_creds):
         """Read settings and create LDAP auth source that can be re-used in
         tests."""
-        super().setUpClass()
-        cls.ldap_user_name = settings.ipa.username_ipa
-        cls.ldap_user_passwd = settings.ipa.password_ipa
-        cls.base_dn = settings.ipa.basedn_ipa
-        cls.group_base_dn = settings.ipa.grpbasedn_ipa
-        cls.ldap_hostname = settings.ipa.hostname_ipa
-        cls.auth = make_ldap_auth_source(
+        base_dn = settings.ipa.basedn_ipa
+        group_base_dn = settings.ipa.grpbasedn_ipa
+        ldap_hostname = settings.ipa.hostname_ipa
+        auth = make_ldap_auth_source(
             {
                 'name': gen_string('alpha'),
                 'onthefly-register': 'true',
-                'host': cls.ldap_hostname,
+                'host': ldap_hostname,
                 'server-type': LDAP_SERVER_TYPE['CLI']['ipa'],
                 'attr-login': LDAP_ATTR['login'],
                 'attr-firstname': LDAP_ATTR['firstname'],
                 'attr-lastname': LDAP_ATTR['surname'],
                 'attr-mail': LDAP_ATTR['mail'],
-                'account': cls.ldap_user_name,
-                'account-password': cls.ldap_user_passwd,
-                'base-dn': cls.base_dn,
-                'groups-base': cls.group_base_dn,
+                'account': module_creds['username'],
+                'account-password': module_creds['password'],
+                'base-dn': base_dn,
+                'groups-base': group_base_dn,
             }
         )
-
-    def setUp(self):
-        """Create new usergroup per each test"""
-        super().setUp()
-        self.user_group = make_usergroup()
-
-    def tearDown(self):
-        """Delete usergroup per each test"""
-        for dict in UserGroup.list():
-            if UserGroup.info({'id': dict['id']})['external-user-groups']:
-                UserGroup.delete({'id': dict['id']})
-        super().tearDown()
-
-    @classmethod
-    @skip_if_not_set('ipa')
-    def tearDownClass(cls):
-        """Delete the IPA auth-source afterwards"""
-        LDAPAuthSource.delete({'id': cls.auth['server']['id']})
-        super().tearDownClass()
+        yield auth
+        LDAPAuthSource.delete({'id': auth['server']['id']})
 
     @pytest.mark.tier2
     @pytest.mark.upgrade
-    def test_positive_create_and_refresh_external_usergroup_with_local_user(self):
+    def test_positive_create_and_refresh_external_usergroup_with_local_user(
```

Comment:
@omkarkhatavkar This one as well to exclude ?

Reply:
Yes looks like we have a similar test here https://github.com/omkarkhatavkar/robottelo/blob/e8c53d9b5e31d6865e1e77062b1162f31726d45c/tests/foreman/cli/test_ldapauthsource.py#L113